### ISSUE TITLE
Error upgrading to Angular 6: Cannot redeclare block-scoped variable 'ngDevMode'

### ISSUE THREAD

**Issue by lippomano** [angular/angular#24165]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[X] Bug report 
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
[ ] Other... Please describe:
</code></pre>

Just update angular from 5.2 to 6.0.3 following [angular](https://update.angular.io/) but
if I try to serve the application  I have this error:

> ERROR in ../node_modules/@angular/core/src/render3/ng_dev_mode.d.ts(9,11): error TS2451: Cannot redeclare block-scoped variable 'ngDevMode'. node_modules/@angular/core/src/render3/ng_dev_mode.d.ts(9,11): error TS2451: Cannot redeclare block-scoped variable 'ngDevMode'.

ng update says it is all updated.

For Tooling issues:
- Node version: v8.9.4
- Platform:  Windows

**Comment by ericmartinezr** [comment]
dup of https://github.com/angular/angular/issues/21925

**Comment by lippomano** [comment]
The advice did not solve my problem

**Comment by alexeagle** [comment]
You need to change your typescript configuration so that you don't resolve duplicate copies of the Angular libraries

**Comment by pcouas** [comment]
I have sam problem, could you precise change for typescript ?

**Comment by fjcunha** [comment]
This tsconfig.json works for me 

`{
  "compilerOptions": {
    "baseUrl": "",
    "allowSyntheticDefaultImports": true,
    "declaration": false,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "lib": [
      "dom",
      "es2015"
    ],
    "module": "es2015",
    "moduleResolution": "node",
    "sourceMap": true,
    "target": "es5",
    "paths": {
      "@angular/*": ["node_modules/@angular/*"]
    }
  },
  "include": [
    "src/**/*.ts",
    "node_modules/ng4-fittext/*.d.ts"
  ],
  "exclude": [
  ],
  "compileOnSave": false,
  "atom": {
    "rewriteTsconfig": false
  }
}`

**Comment by hilkojj** [comment]
@fjcunha 
Thanks it helped me.

Could you explain why this works?

**Comment by fjcunha** [comment]
@hilkojj I'm not sure, but I had to set manually the path "paths": { "@angular/*": ["node_modules/@angular/*"] } to compiler. I think this config will be imported to tsconfig automatically in the future.

**Comment by ctrl-brk** [comment]
Didn't help me. Which tsconfig is that?

**Comment by yohaneslumentut** [comment]
For ionic app, I had to update tsconfig.json at root of my project with:
{
  "compilerOptions": {
    "baseUrl": "",
    ...

    "paths": { 
      "@angular/*": ["node_modules/@angular/*"] 
    }
  },
  ...
}

**Comment by bramdebouvere** [comment]
Thanks @fjcunha , the trick with the paths actually helped me with an issue when referencing local packages that I was stuck on for a couple of hours.

> Error during template compile of 'ModuleName'
  Function calls are not supported in decorators but 'ɵmakeDecorator' was called in 'NgModule'
    'NgModule' calls 'ɵmakeDecorator'.

**Comment by amineb01** [comment]
@fjcunha thank you !

**Comment by smilemuffie** [comment]
@fjcunha Thank you, no warning any more!

**Comment by hoangvu96z** [comment]
@fjcunha it's not working in my project

**Comment by zay-dysania** [comment]
thanks @fjcunha

**Comment by anubhavpsit** [comment]
> @fjcunha
> Thanks it helped me.
> 
> Could you explain why this works?

{
  "compilerOptions": {
    "baseUrl": "./",
    "outDir": "./dist/out-tsc",
    "sourceMap": true,
    "declaration": false,
    "module": "es2015",
    "moduleResolution": "node",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "target": "es5",
    "typeRoots": [
      "node_modules/@types"
    ],
    "lib": [
      "es2017",
      "dom"
    ],
    **"paths": { "@angular/*": ["node_modules/@angular/*"] }** // This is added in new ts config file
  },
  "compileOnSave": false,
}

**Comment by vahiddadaism** [comment]
This happened for me too. Here is what I did to solve it:

- I had a private package which was built using @angular/core version 7.0.4

- I installed it into my project which was using @angular/core version 7.1.0

In this scenario we have two @angular/core with different versions!

I just updated my private package to use @angular/core version 7.1.0, and the problem resolved!


Adding:

> "paths": { 
>   "@angular/*": ["node_modules/@angular/*"] 
> }

does not needed in my scenario.
